Remove need for CGO

nsheridan opened the pull request:
Switch sqlite database driver to one that does not require CGO.

The modernc sqlite driver registers itself with the driver name "sqlite" while the CGO sqlite driver registers itself with the name "sqlite3" and many tools assume this driver name, including the github.com/rubenv/sql-migrate schema migration tool used by cashier.
To acccount for this, register the  (i.e. `sqlite3`) via a `sqlite3_compat` package. 

- Build with CGO_ENABLED=0, but run tests with CGO_ENABLED=1 to enable the race detector.
- Update Go version in README based on go.mod version

Closes https://github.com/cashier-go/cashier/pull/165

nsheridan commented on .github/workflows/ci.yml:
```diff
@@ -23,7 +24,6 @@ jobs:
       - name: start mysql
         uses: shogo82148/actions-setup-mysql@6ab6fe9e7aaea1bb6057a1603872c3a6f644f7a3 # v1.38.1
         with:
-          mysql-version: ${{ matrix.mysql }}
```
Not sure when this was added but we're not matrix testing mysql versions at this time.

SuperQ reviewed (approved):
Looks like `github.com/glebarez/go-sqlite` could use some maintenance love, but otherwise seems OK.

nsheridan commented:
I may have a better solution actually.

nsheridan commented:
I've dropped the `github.com/glebarez/go-sqlite` driver and switched to the `modernc.org/sqlite` driver. I've registered the driver under the `sqlite3` name so that sql-migrate works as expected, no other code changes are required.

SuperQ commented:
Oh, nice.
